### ISSUE TITLE
Ticks that restart their timer on Impulse

### ISSUE THREAD

**Issue by JordanMartinez** [TomasMikula/ReactFX#36]
Coming from TomasMikula/RichTextFX#189, provide an additional `ticks` EventStream that will restart its timer whenever some other EventStream (impulse) emits an event.

**Comment by JordanMartinez** [comment]
One possible way to implement this is to add the following code to `EventStreams`:
````java
public static EventStream<?> ticksThatRestartOn(Duration interval, EventStream<?> impulse) {
    return new EventStreamBase<Void>() {
        private final Timer timer = FxTimer.createPeriodic(
                interval, () -> emit(null));

        @Override
        protected Subscription observeInputs() {
            timer.restart();
            return Subscription.multi(
                    impulse.subscribe(x -> timer.restart()),
                    timer::stop
            );
        }
    };
}
````

**Comment by TomasMikula** [comment]
That would be a correct implementation.

**Comment by TomasMikula** [comment]
Should we call it `restartableTicks`? Although `ticksThatRestartOn` has the advantage that it will be close to `ticks` in Javadoc. A `@see`-also Javadoc tag can compensate for that to some extent.

**Comment by JordanMartinez** [comment]
I see your point. What about `ticksRestartableOn`? It's similar to `conditionOn`
Otherwise, `restartableTicks` does sound better.

**Comment by TomasMikula** [comment]
> Makes sense. However, why is restartableTicks favored over ticksThatRestartOn? Because it is shorter and uses less 'filler' words like "that"? I'm not quite familiar with your naming scheme and your reasoning behind it.

I can't say I have a very consistent naming scheme, but basically `ticksThatRestartOn` feels a bit wordy.

I admit we are in dire need of better documentation.

**Comment by JordanMartinez** [comment]
> I can't say I have a very consistent naming scheme, but basically ticksThatRestartOn feels a bit wordy.
> I admit we are in dire need of better documentation.

Ok. Well, `restartableTicks` sounds good and its shorter. Also, I think the clarification would better serve developers if it were found in the documentation rather than the method name itself.

**Comment by JordanMartinez** [comment]
Once the PR for `ticks0()` gets merged, I'll open up a PR for this issue.

**Comment by JordanMartinez** [comment]
Could you make a new release? Then I can work on fixing the caret visibility issue in RichTextFX

**Comment by TomasMikula** [comment]
I updated the 2.0-SNAPSHOT release that you can use for development.

**Comment by JordanMartinez** [comment]
:+1:

**Comment by JordanMartinez** [comment]
I've updated the `build.gradle` file to use the snapshot version, but it lists the 2.0-M4u1 rather than the snapshot. Is this due to another dependency needing that version's dependency?

**Comment by TomasMikula** [comment]
Ah, yes, that probably will be a problem. I don't know how to force gradle to use the 2.0-SNAPSHOT version, despite some dependency needing 2.0-M4u1, but there might be a way.

**Comment by JordanMartinez** [comment]
I figured out one way. You have to disable the transitive dependencies. Then they'll use the newer version, which shouldn't be an issue in this case:
````groovy
compile (group: 'org.fxmisc.undo', name: 'undofx', version: '1.2') {
    transitive = false
}
compile (group: 'org.fxmisc.flowless', name: 'flowless', version: '0.5-SNAPSHOT') {
    transitive = false
}
````

**Comment by TomasMikula** [comment]
Nice.

**Comment by TomasMikula** [comment]
Resolved by #40.
